pytest-backend/exasol/pytest_backend/__init__.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+from typing import Any
+from datetime import timedelta
+from contextlib import ExitStack
+import ssl
+import pytest
+
+from exasol.saas.client.api_access import (
+    OpenApiAccess,
+    create_saas_client,
+    get_connection_params
+)
+
+from exasol.pytest_itde import bootstrap_db
+
+_BACKEND_ONPREM = 'onprem'
+_BACKEND_SAAS = 'saas'
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--backend",
+        action="append",
+        default=[],
+        help="""List of test backends (onprem, saas). By default, the tests will be
+            run on both backends. To select only one of the backends add the
+            argument --backend=<name-of-the-backend> to the command line. Both backends
+            can be selected like ... --backend=onprem --backend=saas, but this is the
+            same as the default.
+            """,
+    )
+
+
+def pytest_generate_tests(metafunc):
+    if "backend" in metafunc.fixturenames:
+        backend_options = metafunc.config.getoption("--backend")
+        if not backend_options:
+            backend_options = [_BACKEND_ONPREM, _BACKEND_SAAS]
+        metafunc.parametrize("backend", backend_options)
```

Comment:
Have you considered, tested or thought about `indirect`?
https://docs.pytest.org/en/stable/reference/reference.html#pytest.Metafunc.parametrize

Reply:
`indirect` is a way to inject parameters into a fixture from a test. At least that is how I used it. Here I am setting the parameters from a cli.

Reply:
Re: default targets. The motivation for including both backends by default was two-fold.
1. It makes tests super easy.
2. One can argue that the priority is to make sure all tests run on all backends, rather than saving a few seconds of typing. If you need to override the default to include all backends there is a good chance it will be forgotten and the tests on SaaS will never be activated.

Reply:
> `indirect` is a way to inject parameters into a fixture from a test. At least that is how I used it. Here I am setting the parameters from a cli.

AFIK in case of the **_metafunc_**, it is more about when it is executed if I understand it correctly, TL;DR: `collection` vs `setup` - Phase.